### PR TITLE
Update SBT and Scalac

### PR DESCRIPTION
**Pull request by iamrecursion**
### Pull Request Description
This PR updates sbt and scalac, as well as bumping the versions of a few of our dependencies. 

### Important Notes
It fixes the resultant issues.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -207,6 +207,6 @@ branches:
       required_status_checks:
         # Require branches to be up to date before merging.
         strict: true
-        contexts: ["Test Engine (macOS-latest)", "Test Engine (ubuntu-latest)", "Build Engine", "Test Parser (macOS-latest)", "Test Parser (ubuntu-latest)", " Test Parser (windows-latest) ", "license/cla"]
+        contexts: ["Test Engine (macOS-latest)", "Test Engine (ubuntu-latest)", "Test Engine (windows-latest)", "Build Engine", "Test Parser (macOS-latest)", "Test Parser (ubuntu-latest)", " Test Parser (windows-latest) ", "license/cla"]
```
Comment (kustosz) [inline .github/settings.yml]: Make this into a bullet-list

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -26,13 +26,13 @@ jobs:
...
         with:
-          toolchain: nightly-2020-06-09
+          toolchain: nightly
           override: true
-      - name: Install wasm-pack
-        uses: actions-rs/cargo@v1
-        with:
-          command: install
-          args: wasm-pack --version ${{ env.wasmpackVersion }}
+#      - name: Install wasm-pack
```
Comment (kustosz) [inline .github/workflows/rust.yml]: Why is it commented out?

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -148,6 +157,20 @@ git clone https://github.com/enso-org/enso.git
 git clone git@github.com:enso-org/enso.git
 ```
 
+### Getting Set Up
```
Comment (kustosz) [inline docs/CONTRIBUTING.md]: Please mention that it only concerns the new, completely unused, probably-not-what-you're-here-for parser. Particularly that if I understand correctly, it's going to be automated when the parser is actually needed.

File: engine/language-server/src/main/scala/org/enso/languageserver/text/CollaborativeBuffer.scala [hunk 4]
```diff
@@ -307,7 +307,7 @@ class CollaborativeBuffer(
         Some(CapabilityRegistration(CanEdit(bufferPath)))
       else
         None
-    sender ! OpenFileResponse(Right(OpenFileResult(buffer, writeCapability)))
+    sender() ! OpenFileResponse(Right(OpenFileResult(buffer, writeCapability)))
```
Comment (kustosz) [inline engine/language-server/src/main/scala/org/enso/languageserver/text/CollaborativeBuffer.scala:310]: Why? Has something changed in scalac?

File: engine/runtime/src/test/scala/org/enso/compiler/test/core/CorePrimTest.scala [hunk 5]
```diff
@@ -15,6 +17,7 @@ import org.scalatest.matchers.should.Matchers
   * PLEASE NOTE: Many of these tests will be removed once the smart constructors
   * exist.
   */
+@nowarn("cat=deprecation")
```
Comment (kustosz) [inline engine/runtime/src/test/scala/org/enso/compiler/test/core/CorePrimTest.scala:20]: How big is the deprecation issue? Can we not easily fix it?

File: lib/parser-service/src/main/scala/org/enso/parserservice/Server.scala [hunk 6]
```diff
@@ -65,6 +66,7 @@ trait Server {
     *
     * The request's URI is not checked.
     */
+  @nowarn("cat=deprecation")
```
Comment (kustosz) [inline lib/parser-service/src/main/scala/org/enso/parserservice/Server.scala:69]: Again – cannot it be easily fixed?

**Review by kustosz — APPROVED** [review]
Some nitpicks inline

File: engine/language-server/src/main/scala/org/enso/languageserver/text/CollaborativeBuffer.scala (hunk 4 above)
Comment (iamrecursion) [reply engine/language-server/src/main/scala/org/enso/languageserver/text/CollaborativeBuffer.scala:310]: Yep! A method that has a param list must now be called with a param list.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (iamrecursion) [reply .github/workflows/rust.yml]: My mistake. Debugging. Will comment back in.

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (iamrecursion) [reply docs/CONTRIBUTING.md]: Plan is to add the initial bootstrap steps for the Java 11 stuff here as well once radek's PR merges. What do you think?

File: engine/runtime/src/test/scala/org/enso/compiler/test/core/CorePrimTest.scala (hunk 5 above)
Comment (iamrecursion) [reply engine/runtime/src/test/scala/org/enso/compiler/test/core/CorePrimTest.scala:20]: Currently major. Library doesn't export an alternative syntax.

File: lib/parser-service/src/main/scala/org/enso/parserservice/Server.scala (hunk 6 above)
Comment (iamrecursion) [reply lib/parser-service/src/main/scala/org/enso/parserservice/Server.scala:69]: As above.

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (kustosz) [reply docs/CONTRIBUTING.md]: Should be separate sections, one for SBT strangeness, one for "if you want to do rust". Merge after rust is in SBT.

File: engine/language-server/src/main/scala/org/enso/languageserver/text/CollaborativeBuffer.scala (hunk 4 above)
Comment (kustosz) [reply engine/language-server/src/main/scala/org/enso/languageserver/text/CollaborativeBuffer.scala:310]: Thanks, I hate it

**Pull request merged**